components/entity-access-rules/rest/src/main/java/org/phenotips/data/permissions/rest/CollaboratorResource.java

```diff
@@ -35,78 +35,83 @@
 import javax.ws.rs.core.Response;
 
 /**
- * Resource for working with patient record collaborators, one at a time. Collaborators are found by both the internal
- * patient identifier and the internal collaborator identifier.
+ * Resource for working with entity record collaborators, one at a time. Collaborators are found by both the internal
+ * entity identifier and the internal collaborator identifier.
  *
  * @version $Id$
  * @since 1.3M2
  */
-@Path("/patients/{patient-id}/permissions/collaborators/{collaborator-id}")
+@Path("/{entity-type}/{entity-id}/permissions/collaborators/{collaborator-id}")
 @Relation("https://phenotips.org/rel/collaborator")
 @ParentResource(CollaboratorsResource.class)
 @RelatedResources({ PermissionsResource.class, PatientResource.class })
 public interface CollaboratorResource
 {
     /**
-     * Retrieve information about a particular collaborator. If the indicated patient record doesn't exist, or if the
+     * Retrieve information about a particular collaborator. If the indicated entity record doesn't exist, or if the
      * indicated principal is not a collaborator, or if the user sending the request doesn't have the right to view the
-     * target patient record, an error is returned.
+     * target entity record, an error is returned.
      *
-     * @param patientId internal identifier of a patient record
+     * @param entityId internal identifier of an entity record
+     * @param entityType the type of entity
```

Comment:
One has to jump through a lot of code to figure out where to find which type to use (e.g. should it be "patients" or "patient" or what?). Apparently the type is defined and hard-coded in various XXXEntityManager.getType(), and only there, I think there must be a better way to get this information (it was OK for internal use, but should be more clear how to get the string value of the type for the entity one needs once external APIs are involved)

Reply:
Added to the javadoc for now.